### ISSUE TITLE
Using @extend in BEM systax

### ISSUE THREAD

**Issue by kumabotz** [leafo/scssphp#388]
From http://sassmeister.com, SCSS below is compiled into the following css format

``` scss
// scss
.main {
  .block__element {
    font-size: 14px;
    
    &--modifier {
      @extend .block__element;
      font-weight: bold;
    }
  }
}
```

``` scss
// css
.main .block__element, .main .block__element--modifier {
  font-size: 14px;
}

.main .block__element--modifier {
  font-weight: bold;
}
```

But with scssphp v0.6.0 and the SCSS above, I get the error
```
 Fatal error: Maximum function nesting level of '500' reached, aborting! in ../vendor/leafo/scssphp/src/Compiler.php on line 228
```

**Comment by robocoder** [comment]
Please try the fix in 48ccb0d

**Comment by kumabotz** [comment]
It works with that commit, thanks!
